qiskit/providers/ibmq/ibmqbackend.py

```diff
@@ -193,7 +193,10 @@ def retrieve_job(self, job_id):
             IBMQBackendError: if retrieval failed
         """
         try:
-            job_info = self._api.get_status_job(job_id)
+            job_info = self._api.get_job(job_id)
+            if job_info['backend']['name'] != self.name():
+                raise IBMQBackendError('Job "{}" belongs to backend "{}", not "{}".'
```

Comment:
Can you tune the behavior when the job does not belong to the backend, to follow #11 more closely? In particular:
* can you add a `warnings.warn()` between line 197 and 198, that informs the user that the job belongs to another backend (something along a slightly more user-facing version of  `'Job "{}" belongs to backend "{}", not "{}".` - the purpose is to give information to the user so they can correct the situation)
* can you tweak the `IBMQBackendError` message to keep it consistent with the rest of the exceptions raised by the function, as in `'Failed to get job "123": somethingaboutwrongbackend'`?

Reply:
Should there be a difference between the warning and the error message?

Reply:
Ideally, the message for the warning should aim for being informative and guide the user gently to solving the situation, which probably means might be longer - the one in in the exception can be a bit more compact, as it is mostly for developers which hopefully have more insight. Don't fret too much about them!